### DISCUSSION TITLE
Welcome to XyphroLabs-I3C-Saleae-Protocol-Analyzer Discussions!

### DISCUSSION THREAD

**Discussion by xyphro** [xyphro/XyphroLabs-I3C-Saleae-Protocol-Analyzer#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by John-KC-Fu** [comment]
Is it possible to provide a configure file so for re-compling for Linux?

Thanks a lot.

**Comment by xyphro** [comment]
Hi John-KC-Fu,

I have build it for Linux. See attachment of this message. 
[libi3c_analyzer.zip](https://github.com/user-attachments/files/17533141/libi3c_analyzer.zip)

Can you try it out and let me know if it works?

Best regards,

Kai
PS: Unfortunately I don't have MAc OSX, otherwise I'd have it built immediately too :-) My only Mac capable machine is too old to run Mac OSX and runs on Linux nowadays :-)

**Comment by John-KC-Fu** [comment]
Hi Kai,

Thanks a lot for your propt support and it works to me!

Best regards,

John

**Reply by xyphro** [reply]
Great, thanks for feedback! I have also already included it in the github repository.

**Comment by xf-bst** [comment]
Hi xyphro, 
your i3c plugin looks just amazing. I have been looking for this kind of thing. 
One question, why in the position of ack/t-bit/parity, the decoder displays two words, such as CONT+CONT? What is the meaning of these two CONT?
BR
xf

**Reply by xyphro** [reply]
Hi xf,

the ack phase in sdr transfer is different than on i2c. On i2c the 9th bit simply signals acknowledge or no acknowledge.

For i3c, the 9th bit rising and 9th bit falling edge can signal if either master or slave wants to stop the transfer. So 1/2 of bit 9 is controlled by the master and the 2nd half by the slave.

Cont+cont means basically the master wants to continue the transfer and the slave too.

This handshaking stuff really gets more crazy when DDR mode comes into play.

Best regards,

Kai

**Reply by xf-bst** [reply]
Thanks for the quick response!

However, just from the logic trace, one cannot identify whether the master or the slave aborts the transfer, since the 9th is always displayed as a single bit instead of two half bits. 

I just performed a small experiment on my setup, where the controller performs a private read. At the last byte, the controller terminates the transfer. In this case, the decoder still shows CONT+ABORT, but in reality it would be ABORT+CONT. 

I am just wonder it would be enough to just show either CONT or ABORT, since one cannot really differentiate the half bit.

BR,
xf

**Reply by xyphro** [reply]
A read transfer can be terminated with the order cont and abort.

This is when the slave would still have bytes available, but the master still wants to abort and read less bytes.

If you like you can send a waveform + screenshot with  edits in paint to showing what you propose.

I think it makes sense to show things as they are shown now, but that does not mean that other options don't make sense. Maybe I can consider a configuration option that enables diffetent representations.

Just showing an abort or CONtinuation hides who aborts. If you develop a controller or target device this can be useful informstion, but if you develop a higher level sw driver it might not be that useful.

**Reply by xyphro** [reply]
In next refresh round I'll integrate a setting to select the detailness. 
3 levels:
Detailed like now
Summarized as you wish
And a truncated view without termination visibility.
